### PR TITLE
Style as string or struct

### PR DESCRIPTION
**Pull request by wayeast**
Hello. I am using this crate in a project of my own, and had to make the changes encapsulated here for my own purposes. Just wanted to bundle them into a PR to see if they wouldn't be of interest to you.

#### Why
The specification for the object passed to the `Map` constructor in the JS library is documented [here](https://docs.mapbox.com/mapbox-gl-js/api/map/) (this is the `MapOptions` struct in this crate). In the JS spec, the `style` property can be an `Object` conforming to the [Mapbox Style spec](https://docs.mapbox.com/style-spec/reference/) or a `string` that references a document with such a spec. Currently, this crate only accepts the string form of this property. This PR adds the ability for a style spec `Object` to be passed as well.

#### What
The core of this PR is the addition of a `StyleOrRef` enum that expresses the optional `Object` or `string` in JS. It also adds the minimal features of a valid `Style` object (the required properties of the above-linked spec). It also makes some small modifications to existing method definitions to support the new type. Last, it adds a simple example to demonstrate how to use the new enum type.

#### NB
1. As mentioned above, this PR only adds support for the minimal, required properties of a `Style` object -- it does not pretend to fully support these objects! If this PR is accepted, it should be followed up with another PR to add all of that support.
2. There are other places in the object passed to a `Map` constructor that allow for a similar option between a simple type or more elaborate object. This PR is only interested in the `style` property; if it is of interest to add similar functionality elsewhere, I think this PR may serve as a good pattern to adopt.

### CODE REVIEW

**Comment by yukinarit** [comment]
Hi @wayeast Thanks for the suggestion! Let me take a look on this weekend!

File: src/lib.rs [hunk 1]
```diff
@@ -288,7 +286,7 @@ impl MapOptions {
         }
     }
 
-    pub fn style(mut self, style: String) -> MapOptions {
+    pub fn style(mut self, style: StyleOrRef) -> MapOptions {
```
Comment (yukinarit) [inline src/lib.rs:289]: I slightly prefer providing two functions to single function with enum.

```rust
pub fn style_ref(mut self, style_ref: String);
pub fn style(mut self, style: Style);
```

**Review by yukinarit — COMMENTED** [review]
I left a comment, PTAL :pray:

File: src/lib.rs (hunk 1 above)
Comment (wayeast) [reply src/lib.rs]: Yeah, I like your suggestion as well. Have added a new commit to the PR to bring it into line.

**Review by yukinarit — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**